src/Execution/ExecutionStrategy.php

```diff
@@ -767,12 +776,12 @@ private function collectAndExecuteSubFields(
                     $returnType,
                     $fieldNode->getSelectionSet(),
                     $subFields,
-                    new \ArrayObject()
+                    $visitedFragmentNames
                 );
             }
         }
 
-        if ($subFields->count()) {
+        if (count($subFields) > 0) {
```

Comment:
Can't we use `!empty($subFields)`?

Reply:
`if(count($subFields))` also works. But this for clarity.